src/pages/MergePositions/Contents.tsx

```diff
@@ -164,74 +194,234 @@ export const Contents = () => {
+    return {
+      CollateralValue: {
+        type: null,
+        value: null,
+      },
+      ToCreationDate: null,
+      FromCreationDate: null,
+      TextToSearch: {
+        type: PositionSearchOptions.All,
+        value: null,
+      },
+      WrappedCollateral: WrappedCollateralOptions.All,
+    }
+  }, [])
+
+  const { data: positions, error: errorPositions } = usePositionsList(advancedFilters)
+
+  const onRowClicked = useCallback(
+    async (position: PositionWithUserBalanceWithDecimals) => {
+      setPosition(position)
+
+      setMergeablePositions([])
+      setConditionIds([])
+
+      if (positions && positions.length > 0 && !!position) {
+        setIsLoadingConditions(true)
+        setIsLoadingMergeablePositions(true)
+
+        const positionsFiltered = positions.filter(
+          (positionWithBalance: PositionWithUserBalanceWithDecimals) =>
+            //#1 Filter, only positions with balance
+            !positionWithBalance.userBalanceERC1155.isZero() &&
+            //#2 Filter, remove original selected position
+            positionWithBalance.id.toLowerCase() !== position.id.toLowerCase() &&
+            //#3 Filter, positions with the same condition set
+            positionWithBalance.conditionIds.sort().join('') ===
+              position.conditionIds.sort().join('') &&
+            //#4 Filter, positions with the same collateral
+            positionWithBalance.collateralToken.toLowerCase() ===
+              position.collateralToken.toLowerCase() &&
+            //#5 Filter, allow only disjoint positions
+            positionWithBalance.conditions.filter((condition: ConditionInformation) =>
+              isDisjointPartition(positionWithBalance.indexSets, condition.outcomeSlotCount)
+            ).length > 0
+        )
+
+        const positionsPromises = positionsFiltered.map(async (positionFiltered) => {
+          const { collateralToken, conditionIds, indexSets } = positionFiltered
+          const balanceOfPositionId = await CTService.balanceOf(positionFiltered.id)
+          const erc20Service = new ERC20Service(provider, collateralToken)
+          const tokenPosition = await erc20Service.getProfileSummary()
+
+          return {
+            position: positionFiltered,
+            positionPreview: positionString(
+              conditionIds,
+              indexSets,
+              balanceOfPositionId,
+              tokenPosition
+            ),
+          }
+        })
+
+        const possibleMergeablePositions: MergeablePosition[] = await Promise.all(positionsPromises)
+
+        // Calculate condition Ids to fulfill the dropdown
+        const conditionIds: string[] = possibleMergeablePositions.reduce(
+          (acc: string[], cur: MergeablePosition) => acc.concat([...cur.position.conditionIds]),
+          []
+        )
+
+        // Remove duplicates
+        const possibleConditions = conditionIds.filter(
+          (item, pos) => conditionIds.indexOf(item) === pos
+        )
```

Comment:
Every position that passes the filter should have the same set of conditionsIds, so it should be enough with using position.conditionsIds or something like that. `lodash.uniq` is available too.